### DISCUSSION TITLE
Include HTTPExceptions in OpenAPI/Swagger that are raised in a dependency

### DISCUSSION THREAD

**Discussion by p-rinnert** [fastapi/fastapi#9127] [Questions]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

```Python
from fastapi import Depends, FastAPI, Query, HTTPException, status

app = FastAPI()


def dependency(name: str = Query(...)):
    if name == 'test':
        raise HTTPException(status_code=status.HTTP_406_NOT_ACCEPTABLE)
    return name


@app.get(path='/example', responses={"403": {"description": "Forbidden name was used"}})
def example(name: str = Depends(dependency)):

    if name == 'forbidden name':
        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN)

    return name
```

### Description
In our project we use dependencies that are reused in multiple endpoints. 
The dependencies can raise exceptions and we would like to be able to see these exceptions documented in OpenAPI/Swagger.
If an exception is raised inside an endpoint (like in `example()` above) we add the raised exception to OpenAPI with the `responses={}` in the path operation decorator as described in [the documentation.](https://fastapi.tiangolo.com/advanced/additional-responses/)

Is there a way to add a similar `responses={}` to the dependency?
The only way we found is to manually go through the dependencies in each endpoint (and the dependencies used by the dependencies) and check for raised exceptions.
This seems very complicated and hard to maintain but maybe there is an less repetetive way?
Is it possible to add a `responses={}` to a whole app or router, so these codes are shown in all routes? (like `app=FastAPI( responses={"403": {"description": "foo"}})`)

This issue might be related to  #https://github.com/tiangolo/fastapi/issues/1999

**Comment by ycd** [comment]
Does this work in your case? 

```python
responses = {
    404: {"description": "Item not found"},
    302: {"description": "The item was moved"},
    403: {"description": "Forbidden name was used"},
}

@app.get("/dummy", responses={**responses})


```
